tools/sil-opt/SILOpt.cpp

```diff
@@ -202,6 +202,43 @@ AssumeUnqualifiedOwnershipWhenParsing(
     "assume-parsing-unqualified-ownership-sil", llvm::cl::Hidden, llvm::cl::init(false),
     llvm::cl::desc("Assume all parsed functions have unqualified ownership"));
 
+/// \brief Regular expression corresponding to the value given in one of the
+/// -pass-remarks* command line flags. Passes whose name matches this regexp
+/// will emit a diagnostic.
+struct PassRemarksOpt {
+  std::shared_ptr<llvm::Regex> Pattern;
+
+  void operator=(const std::string &Val) {
+    // Create a regexp object to match pass names for emitOptimizationRemark.
+    if (!Val.empty()) {
+      Pattern = std::make_shared<llvm::Regex>(Val);
+      std::string RegexError;
+      if (!Pattern->isValid(RegexError))
+        llvm::report_fatal_error("Invalid regular expression '" + Val +
+                                     "' in -sil-remarks: " + RegexError,
+                                 false);
+    }
+  }
+};
```

Comment:
Why do this ahead of time rather than using a normal string option and computing it later?